search/mongo_search.go

```diff
@@ -455,7 +455,7 @@ func (m *MongoSearcher) createQuantityQueryObject(q *QuantityParam) bson.M {
 				bson.M{"unit": ci(q.Code)},
 			}
 		} else {
-			criteria["code"] = ci(q.Code)
+			criteria["code"] = q.Code
 			criteria["system"] = ci(q.System)
```

Comment:
As discussed with @cjduffett, leaving `system` as case-insensitive since URIs are generally considered to be case-insensitive.  This means that searches which specify a system will still suffer a performance hit, but searches without systems will be fast.